docs/xenit-style-guide/golang.md

```diff
@@ -132,52 +136,130 @@ case args.Push != nil:
-    // We use zap and zapr here, but you can really use any of the loggers
-    // supported by logr
-    zl, _ := zap.NewProduction()
-    logger := zapr.NewLogger(zl)
-
-    // Add a ginlogr middleware, which:
-    //   - Logs all requests, like a combined access and error log.
-    //   - Logs to stdout.
-    //   - RFC3339 with UTC time format.
-    r.Use(ginlogr.Ginlogr(logger, time.RFC3339, true))
-
-    // Logs all panic to error log
-    //   - RFC3389 with UTC time format.
-    //   - stack means whether output the stack info.
-    r.Use(ginlogr.RecoveryWithLogr(logger, time.RFC3339, true, true))
-
-    // Example ping request.
-    r.GET("/ping", func(c *gin.Context) {
-        c.String(200, "pong "+fmt.Sprint(time.Now().Unix()))
-    })
+	zapLog, err := zap.NewProduction()
+	if err != nil {
+		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
+	}
+	log := zapr.NewLogger(zapLog)
+	log.Info("Logr in action!", "the answer", 42)
+}
+```
 
-    // Example when panic happen.
-    r.GET("/panic", func(c *gin.Context) {
-        panic("An unexpected error happen!")
-    })
+The logging object should be passed through the context when calling a function that needs to log. This removes the need to add an additional parameter that needs to be tracked.
+
+```golang
+ctx := logr.NewContext(context.Background(), log)
+run(ctx)
 
-    // Listen and Server in 0.0.0.0:8080
-    r.Run(":8080")
+func run(ctx context.Context) {
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("running function")
 }
 ```
 
-### Datadog
+## HTTP
 
-Using the `status` field when logging can interfere with how Datadog interprets the severity of the log. If you have a log with fields `status = 400` and `level = error`, then Datadog reports this as an info log. If you use another field for the status code, for example `http_response.status_code`, together with `level = error`, Datadog will report it as an error log. You can read more about the `status` field and other reserved log fields in Datadog [here](https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#reserved-attributes).
+The Go standard library includes a http package that works well for simple applications, but requires a lot of custom code when building larger projects. While it is fine to just use the standard library for simple applications it is preferable to switch to [Gin](https://github.com/gin-gonic/gin) as project feature requirements develop. Gin provides extra functionality and extensions to simplify things like parsing parameters in the URL path and endpoint authorization.
+
+The HTTP server should be started and stopped in accordance to the Startup and Shutdown documentation. The HTTP server should be gracefully stopped before the program exits to make sure that all in flight requests are processed. A good practice is to use the [logger middleware](https://github.com/XenitAB/pkg/tree/main/middleware/logger) to log all incoming requests to the HTTP server.
```

Comment:
Do you want to say something about why graceful shutdown is needed?

E.g. usually an application gets a signal which gives it time to close e.g. database connections. If an application is exited too abruptly it might leak connections or corrupt state. Something along those lines?